pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1060,6 +1060,40 @@ func (m *ManagerImpl) isDevicePluginResource(resource string) bool {
 	return false
 }
 
+// GetAllDevices returns information about all the devices known to the manager
+func (m *ManagerImpl) GetAllDevices() podresourcesapi.ResourceDeviceMap {
+	m.mutex.Lock()
+	resp := make(podresourcesapi.ResourceDeviceMap)
+	for resourceName, resourceDevs := range m.allDevices {
+		resp[resourceName] = make(podresourcesapi.DeviceMap)
+		for devID, dev := range resourceDevs {
+			cntDev := podresourcesapi.ContainerDevices{
+				ResourceName: resourceName,
+				DeviceIds:    []string{devID},
+			}
+
+			// Get*() accessors are safe if intermediate pointers are nil
+			if nodes := dev.GetTopology().GetNodes(); nodes != nil {
+				cntDev.Topology = &podresourcesapi.TopologyInfo{
+					Nodes: []*podresourcesapi.NUMANode{
+						// note: returns 0 if NUMANode* is nil (should not happen here)
+						// note: each pluginapi.Device has a single ID but can have
+						// a slice of NUMANodes on its TopologyInfo. Since the same ID
+						// cannot be on multiple NUMA Node, we just pick the first value.
+						{ID: nodes[0].GetID()},
```

Comment:
that is wrong assumption. One device ID can have multiple "numa" node ids.
Device is connected to PCI bus. PCI bus is a property of CPU die (or e.g. I/O die in AMD Rome). this die can span multiple "numa" nodes, as NUMA id represent only memory controller number and not guarantee 1-to-1 map to e.g. PCI bus.

Reply:
Thanks for pointing this out. This indeed voids the assumption. So we can have (and thus we should report) the same device ID in multiple NUMA nodes, right?

Reply:
better not to use numa id as a key at all. ideally it should be presented in "zone" level as we discussed in scheduler extension crds, but as we don't have those now, for this particular API if we want to express internal representation of TopologyManager that using memory controller numbers as "zone id", then yes, announce same device multiple times for each ID, according to topology hint received from device plugin.

Reply:
ack, thanks for pointing this out. This indeed something I for myself need to fully absorb (I'm used to simpler topologies) so thanks for taking the time to explain it again.
For the purposes of this PR, indeed we reuse the same internal representation Topology Manager uses, but I'll strive to take your points into account. I'll post an update soon.

Reply:
done in https://github.com/kubernetes/kubernetes/pull/95734/commits/521ee88a0f3142650eb235e8ea93b8e6e825f773